tests/ui/attributes/mixed_export_name_and_no_mangle.rs

```diff
@@ -0,0 +1,17 @@
+// issue: rust-lang/rust#47446
+//@ run-rustfix
+//@ check-pass
+
+#![warn(unused_attributes)]
+#[no_mangle]
+//~^ WARN the `#[no_mangle]` attribute may not be used in combination with `#[export_name]`
+#[export_name = "foo"]
+pub fn bar() {}
+
+#[unsafe(no_mangle)]
```

Comment:
@Urgau maybe you have an idea why this `unsafe` wrapper is not showing up in the error message?

Reply:
That's because you are using `to_ident_string()` which only look at the identifier but an attribute is composed of many more things than a single identifier.

For good reproduction I would suggest using the AST pretty printer (crate `rustc_ast_pretty`), [rustc_ast_pretty::pprust::attribute_to_string](https://doc.rust-lang.org/stable/nightly-rustc/rustc_ast_pretty/pprust/fn.attribute_to_string.html), but looking at the code I don't thing it handles safety yet (which is a bug).

I would suggest leaving for now the safety part out and having a `// FIXME: Handle attribute unsafety` somewhere in the code.

Alternatively you could already add the call to the pretty printer, add the fixme, and if you feel like it fix the pretty-printer in another PR.

But in anycase, let's not block the PR on this, let's make CI pass, so we can resolve the concern.

Reply:
Thanks for your suggestion, I've used now the pretty printer, the good thing it prints already the full attributes with the `#[]` surroundings,

But unfortunately the `unsafe()` encapsulation is still not showing up.

Reply:
It's probably around here, https://github.com/rust-lang/rust/blob/3ec4308f6cb4bba3140d736d9ebd53b234fa7df8/compiler/rustc_ast_pretty/src/pprust/state.rs#L628-L630, if you want you can try fixing it (in another pr), but in the mean time let's move on this PR with a `FIXME`, so we can unblock the concern.

Reply:
Opened #132180 to fix the pretty-printing bug.

Reply:
@Urgau, thanks to your fix, the error message now adds up to how it should. So, I guess everything is addressed and solved now.